### DISCUSSION TITLE
[SOLVED] Does Capacitor have Debugging/Dev tools? (similar to React-Native's Dev tools for example)

### DISCUSSION THREAD

**Discussion by orjwan98** [ionic-team/capacitor#4691] [Q&A]
I am new to Capacitor Native. I started working with it a week ago, and though I swept through the docs and the internet looking for debuggers/devtools for Capacitor, I didn't find anything. For context, I am using Capacitor Native with CRA and aside from using XCode's and AndroidStudio's built-in consoles I have no way of debugging things. I am pretty much developing in the dark and it's very frustrating. 

Can someone point me in the right direction please? Maybe I am missing something. Thank you in advance!

**Comment by jcesarmobile** [comment]
You can debug them as as if they were websites.

In android you should do it with chrome remote inspector, you type `chrome://inspect/` in your desktop chrome browser and the plugged device should appear there and show the apps that are open.

In iOS you should use safari remote inspector. You have to enable the Development menu in Safari desktop and the plugged devices (and even wireless now) should appear there.

**Comment by mlynch** [comment]
Like @jcesarmobile said, Capacitor is great because you can build and debug your app like any other website, so there's no need for any custom debugging tools like in the React Native and other worlds.

One other tool to mention is [inspect.dev](https://inspect.dev) which is a nicer experience than Safari dev for remotely debugging your Capacitor app, definitely recommend checking that out and people in the community seem to really like it.

**Comment by orjwan98** [comment]
Thank you so much @jcesarmobile & @mlynch for answering me. Your answers are exactly what I've been looking for. I've been struggling with this for a week now. You have no idea how thankful I am you responded. Btw, one more question, is this mentioned anywhere in the docs and I just seemed to have missed it?
